src/masternodes/rpc_masternodes.cpp

```diff
@@ -41,17 +41,17 @@ UniValue mnToJSON(uint256 const & nodeId, CMasternode const& node, bool verbose,
 
         // Only get targetMultiplier for active masternodes
         if (node.IsActive()) {
-            auto usedHeight = currentHeight <= Params().GetConsensus().EunosHeight ? node.creationHeight : currentHeight;
-            auto stakerBlockTime = pcustomcsview->GetMasternodeLastBlockTime(node.operatorAuthAddress, usedHeight);
-            // No record. No stake blocks or post-fork createmastnode TX, use fork time.
-            if (!stakerBlockTime) {
-                if (auto block = ::ChainActive()[Params().GetConsensus().DakotaCrescentHeight]) {
-                    stakerBlockTime = std::min(GetTime() - block->GetBlockTime(), Params().GetConsensus().pos.nStakeMaxAge);
+            // Get block times with next block as height
+            const auto subNodesBlockTime = pcustomcsview->GetBlockTimes(node.operatorAuthAddress, currentHeight + 1, node.creationHeight, timelock);
+
+            if (currentHeight >= Params().GetConsensus().EunosPayaHeight) {
+                const uint8_t loops = timelock == CMasternode::TENYEAR ? 4 : timelock == CMasternode::FIVEYEAR ? 3 : 2;
+                for (uint8_t i{0}; i < loops; ++i) {
+                    obj.pushKV(strprintf("multiplierSubnode%d", i), pos::CalcCoinDayWeight(Params().GetConsensus(), GetTime(), subNodesBlockTime[i]).getdouble());
```

Comment:
Can we make it as 2nd layer array? i.e. 

```js
multipliers: [12, 1, 42]
```

This is much better because you know exactly how many multipliers/subnodes you get, and a consumer can iterate through easily.

Reply:
That's is a much better idea.